app/models/specific_tube_rack_creation.rb

```diff
@@ -0,0 +1,436 @@
+  #   - :tube_rack_purpose_uuid [String] The UUID of the tube rack purpose.
+  #   - :tube_rack_name [String] The name of the tube rack.
+  #   - :tube_rack_barcode [String] The barcode of the tube rack.
+  #   - :tube_rack_metadata_key [String] The metadata key for the tube rack.
+  #   - :racked_tubes [Array<Hash>] An array of hashes defining the tubes to be created within the tube rack.
+  #
+  # @raise [StandardError] if any of the tube rack creation processes fail.
+  #
+  # @return [TubeRack] The created tube rack.
+  def create_child_tube_rack(rack_attributes)
+    child_purpose = find_tube_rack_purpose(rack_attributes[:tube_rack_purpose_uuid])
+    child_purposes << child_purpose
+
+    new_tube_rack = child_purpose.create!(name: rack_attributes[:tube_rack_name])
+    handle_tube_rack_barcode(rack_attributes[:tube_rack_barcode], new_tube_rack)
+    add_tube_rack_metadata(rack_attributes[:tube_rack_metadata_key], rack_attributes[:tube_rack_barcode], new_tube_rack)
+
+    create_racked_tubes(rack_attributes[:racked_tubes], new_tube_rack)
+
+    new_tube_rack
+  end
+
+  # Finds the tube rack purpose using the provided UUID.
+  #
+  # This method queries the TubeRack::Purpose model to find the first record
+  # that matches the given UUID. If a matching record is found, it is returned.
+  # If no matching record is found, an error is raised.
+  #
+  # @param [String] uuid The UUID of the tube rack purpose to find.
+  #
+  # @raise [StandardError] if no matching TubeRack::Purpose record is found.
+  #
+  # @return [TubeRack::Purpose] The found TubeRack::Purpose object.
+  def find_tube_rack_purpose(uuid)
+    tr_purpose = TubeRack::Purpose.with_uuid(uuid).first
+    return tr_purpose if tr_purpose
+
+    error_message = "The tube rack purpose with UUID '#{uuid}' was not found."
+    raise StandardError, error_message
+  end
+
+  # Handles the barcode assignment for a new tube rack.
+  #
+  # This method checks if a barcode already exists for the given tube rack barcode.
+  # If the barcode does not exist, it creates a new barcode for the tube rack.
+  # If the barcode already exists, it redirects the existing barcode to the new tube rack instance.
+  # This is done to allow the re-use of the physical tube rack, which has an etched barcode.
+  #
+  # @param [String] tube_rack_barcode The barcode of the tube rack.
+  # @param [TubeRack] new_tube_rack The new tube rack object to which the barcode will be assigned.
+  #
+  # @raise [StandardError] if the barcode cannot be created or redirected.
+  #
+  # @return [void]
+  def handle_tube_rack_barcode(tube_rack_barcode, new_tube_rack)
+    existing_barcode_record = Barcode.includes(:asset).find_by(barcode: tube_rack_barcode)
+
+    if existing_barcode_record.nil?
+      create_new_barcode(tube_rack_barcode, new_tube_rack)
+    else
+      redirect_existing_barcode(existing_barcode_record, new_tube_rack, tube_rack_barcode)
+    end
+  end
```

Comment:
This (handle_tube_rack_barcode) feels like a core method that should probably be elsewhere - on the tube rack model, or purpose perhaps 🤔

Reply:
yeah I agree, too much code in this labware creator. can discuss.

Reply:
@KatyTaylor  Create refactor story to create central tube rack creation code to be shared between this and sample manifest upload code.